api/src/main/java/org/apache/iceberg/actions/RewriteTablePath.java

```diff
@@ -112,5 +122,15 @@ interface Result {
 
     /** Name of latest metadata file version */
     String latestVersion();
+
+    /** count of rewrite delete files, default value is 0 */
+    default int deleteFilesCount() {
+      return 0;
+    }
+
+    /** count of rewrite metadata files involved, default value is 0 */
+    default int metadataFilesCount() {
+      return 0;
```

Comment:
Can you share a bit more on what are those 2 results are used for? I think add some unit tests with non-zero results would help.